orc8r/cloud/go/services/certifier/tokens.go

```diff
@@ -0,0 +1,102 @@
+package certifier
+
+import (
+	"crypto/rand"
+	"errors"
+	"fmt"
+	"hash/crc32"
+	"strings"
+
+	"github.com/jxskiss/base62"
+)
+
+type TokenType string
+
+const (
+	Personal TokenType = "op"
+)
+
+// Length of checksum of token's byte array
+const checksumLength = 4
+
+// GenerateToken generates a random 32-byte token with a checksum
+func GenerateToken(tokenType TokenType) (string, error) {
+	// Generate 32 random bytes
+	bytes := make([]byte, 32)
+	_, err := rand.Read(bytes)
+	if err != nil {
+		return "", err
+	}
+
+	// Generate checksum bytes
+	checksum := crc32.ChecksumIEEE(bytes)
+	checksumBytes := i32tob(checksum)
+
+	// Combine to form final token
+	finalBytes := append(bytes, checksumBytes...)
+	token := base62.StdEncoding.EncodeToString(finalBytes)
+	return fmt.Sprintf("%v_%s", tokenType, token), nil
+}
+
+// ValidateToken makes sure the token has the appropriate header and
+// that the checksum is correct
+func ValidateToken(token string) error {
+	value, err := stripTokenHeader(token)
+	if err != nil {
+		return err
+	}
+	if err := validateTokenChecksum(value); err != nil {
+		return err
+	}
+	return nil
+}
+
+func stripTokenHeader(token string) (string, error) {
+	s := strings.Split(token, "_")
+	if len(s) != 2 {
+		return "", errors.New("missing token type")
+	}
+	tokenType, value := s[0], s[1]
+
+	// Validate token tokenType
+	switch tokenType {
+	case string(Personal):
```

Comment:
Nit: move the type cast to the tokenType

Reply:
Wdym here?